srcCpp/RTIDDSImpl.cxx

```diff
@@ -1334,6 +1540,49 @@ class RTIDynamicDataPublisher : public IMessagingWriter
             perftest_cpp::MilliSleep(nsec / 1000000);
         }
     }
+#ifdef RTI_CUSTOM_TYPE
+  private:
+    bool is_setinel_size(int size) {
+        return size == perftest_cpp::INITIALIZE_SIZE
+                || size == perftest_cpp::FINISHED_SIZE
+                || size == perftest_cpp::LENGTH_CHANGED_SIZE
+                || size == 0;
+    }
+
+    bool get_serialize_size_custom_type_data(unsigned int &size) {
+        bool success = true;
+        char *buffer = NULL;
+        DDS_ReturnCode_t retcode = data.to_cdr_buffer(
+                NULL,
+                (unsigned int &)size);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "to_cdr_buffer failed: %d.\n", retcode);
+            success = false;
+        }
+        RTIOsapiHeap_allocateBufferAligned(
+                &buffer,
+                size,
+                RTIOsapiAlignment_getAlignmentOf(void *));
```

Comment:
RTIOsapiAlignment_getAlignmentOf(char *)
`char *` and `void *`  have the same size, so it's not a problem.
But maybe you want to set the same type as the `buffer`.

Reply:
Done